File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        numcpy = x
        reverse = 0
        isNegative = False
        if numcpy < 0:
            numcpy *= -1
            isNegative = True
        while(numcpy >= 10):
            to_add = numcpy % 10
            numcpy = numcpy // 10
            reverse = reverse * 10 + to_add
        reverse = reverse * 10 + (numcpy % 10)
        if reverse > 4294967295:
            return 0
        if reverse >= 2147483651:
            return 0
        if isNegative:
            return reverse * -1
        return reverse

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(long int x) {
        long int reverse = 0;
        bool isNegative = false;
        if(x < 0)
        {
            x *= -1;
            isNegative = true;
        }
        while(x >= 10)
        {
            long int to_add = x % 10;
            x /= 10;
            reverse = reverse * 10 + to_add;
        }
        reverse = reverse * 10 + (x%10);
        if(reverse > 4294967295 || reverse >= 2147483651)
        {
            return 0;
        }
        if(isNegative)
        {
            reverse *= -1;
        }
        return reverse;
    }
};